File: first-unique-char-in-string.rb
# @param {String} s
# @return {Integer}
def first_uniq_char(s)
    chars = Hash.new(0)

    s.each_char do |str|
        chars[str] = chars[str] + 1
    end

    s.each_char.with_index do |str, i|
        return i if chars[str] == 1
    end

    return -1
end

File: first-unique-char-in-string.ts
function firstUniqChar(s: string): number {
    let chars = {}

    for (let idx = 0; idx < s.length; idx++) {
        if (chars[s[idx]] === undefined) {
            chars[s[idx]] = 1;
        } else {
            chars[s[idx]] += 1;
        }
    }

    for (let idx = 0; idx < s.length; idx++) {
        if (chars[s[idx]] === 1) {
            return idx;
        }
    }

    return -1;
}